### PR TITLE
[AzureCommunicationCommon]ObjC tests for CommunicationTokenCredential

### PR DESCRIPTION
**Pull request by AikoBB**
# Description
Added all missing objC tests for `CommunicationTokenCredential`, covered with the tests of the implementation of the [User Story 3074675](https://skype.visualstudio.com/SPOOL/_workitems/edit/3074675).

### CODE REVIEW

File: sdk/communication/AzureCommunicationCommon/Tests/ObjC/ObjCAutoRefreshProactiveTokenCredentialTests.m [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+    CommunicationTokenCredential *credential = [[CommunicationTokenCredential alloc]
+                                                initWithOptions: tokenRefreshOptions
+                                                error:nil];
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(1 * NSEC_PER_SEC)), dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
+        [credential tokenWithCompletionHandler:^(CommunicationAccessToken * _Nullable accessToken,
+                                                 NSError * _Nullable error) {
+            XCTAssertNotNil(error);
+            XCTAssertEqual([error.localizedDescription containsString: errorDesc], YES);
+            XCTAssertNil(accessToken);
+            XCTAssertEqual(weakSelf.fetchTokenCallCount, 2);
```
Comment (petrsvihlik) [inline sdk/communication/AzureCommunicationCommon/Tests/ObjC/ObjCAutoRefreshProactiveTokenCredentialTests.m:157]: Why is this "2"? Please remind me...

File: sdk/communication/AzureCommunicationCommon/Tests/ObjC/ObjCAutoRefreshProactiveTokenCredentialTests.m [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    [self waitForExpectationsWithTimeout:self.timeout handler:^(NSError * _Nullable error) {
+        [self failForTimeout:error testName:expectation.expectationDescription];
+    }];
+}
+
+- (void)test_ShouldBeCalledIfTokenExpiringSoon {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"test_ShouldBeCalledIfTokenExpiringSoon"];
+    
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+    int minutes[] = {1, 9};
```
Comment (petrsvihlik) [inline sdk/communication/AzureCommunicationCommon/Tests/ObjC/ObjCAutoRefreshProactiveTokenCredentialTests.m:101]: I suppose anything above 9 is covered by `test_ShouldNotBeCalledBeforeExpiringTime`.

Comment (AikoBB) [reply sdk/communication/AzureCommunicationCommon/Tests/ObjC/ObjCAutoRefreshProactiveTokenCredentialTests.m:101]: yes

File: sdk/communication/AzureCommunicationCommon/Tests/ObjC/ObjCAutoRefreshProactiveTokenCredentialTests.m [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        });
+        dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);
+    }
+    
+    [self waitForExpectationsWithTimeout:self.timeout handler:^(NSError * _Nullable error) {
+        [self failForTimeout:error testName:expectation.expectationDescription];
+    }];
+}
+
+- (void)test_FetchTokenReturnsError {
```
[the remark this answers is not in the document]
Comment (AikoBB) [reply sdk/communication/AzureCommunicationCommon/Tests/ObjC/ObjCAutoRefreshProactiveTokenCredentialTests.m:133]: Does `test_ShouldThrowExceptionWhenTokenRefresherThrowsException` sound better?

File: sdk/communication/AzureCommunicationCommon/Tests/ObjC/ObjCAutoRefreshProactiveTokenCredentialTests.m (hunk 1 above)
Comment (AikoBB) [reply sdk/communication/AzureCommunicationCommon/Tests/ObjC/ObjCAutoRefreshProactiveTokenCredentialTests.m:157]: As an expired token was passed as the initial token in the constructor refreshing token will be triggered and finalized with an error, the second time when the token API is called in line 152

File: sdk/communication/AzureCommunicationCommon/Tests/ObjC/ObjCAutoRefreshProactiveTokenCredentialTests.m (hunk 3 above)
[the remark this answers is not in the document]
Comment (petrsvihlik) [reply sdk/communication/AzureCommunicationCommon/Tests/ObjC/ObjCAutoRefreshProactiveTokenCredentialTests.m:133]: Makes much more sense! :) If you want to make it shorter, go with `test_ShouldThrowExceptionWhenTokenRefresherThrows` or even `test_ShouldThrowWhenTokenRefresherThrows`

**Review by petrsvihlik — APPROVED** [review]

**Pull request merged**